### PR TITLE
docs: create Renovate and changelogs page

### LINKED ISSUE #25874: Creating a new "Renovate and changelogs" page in the docs
### Discussed in https://github.com/renovatebot/renovate/discussions/25867

<div type='discussions-op-text'>

<sup>Originally posted by **HonkingGoose** November 19, 2023</sup>
### Background

Many users ask about how Renovate fetches changelogs. The maintainers of course answer most of those questions, but we should get those answers and concepts into the docs.

### Current info

The Renovate docs mention changelogs in a few places:

- [Private packages, looking up changelogs](https://docs.renovatebot.com/getting-started/private-packages/#looking-up-changelogs)
- [Maven datasource, making your changelogs fetchable](https://docs.renovatebot.com/modules/datasource/maven/#making-your-changelogs-fetchable)
- [Github.com token for changelogs](https://docs.renovatebot.com/getting-started/running/#githubcom-token-for-changelogs)
- [How to get Renovate to see Docker image changelogs, as image maintainer](https://docs.renovatebot.com/modules/datasource/docker/)
- [`fetchChangeLogs` config option](https://docs.renovatebot.com/configuration-options/#fetchchangelogs)
- [`customChangelogUrl` config option](https://docs.renovatebot.com/configuration-options/#customchangelogurl)
- Some Renovate managers have a "Renovate fetches changelogs step" in the how it works sections, but most do _not_ mention how to get changelogs visible to Renovate

### Related issue:

- #14138

### Discussion

It would be nice to have a single page that explains common questions and concepts, like:

- How Renovate "sees" changelogs
- From which platforms can Renovate fetch changelogs
- Troubleshooting missing changelogs
- Advice for package maintainers (do `x` to get Renovate to see changelogs)
  - npm package advice
  - yarn package advice
  - other "big/much used" package advice
- Apparently Renovate can read a `changelog.json` file, again JSON, I don't think we explain that at all...
- Other things I'm forgetting to list, feel free to suggest!</div>

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Create "Renovate and changelogs page" that should explain:
  - How Renovate fetches changelogs
  - How maintainers can ensure Renovate sees their changelogs, for the commonly used package ecosystems
  - Troubleshooting missing changelogs
  - Relevant config options to configure changelog fetching behavior

## Context

Closes #25874.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/key-concepts/changelogs.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+Explain how to get Nuget changelogs here.
+
+### Go packages
+
+Explain how to get Go changelogs here.
+
+### Ruby packages
+
+Explain how to get Ruby (gem) changelogs here.
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/changelogs.md:83]:
```suggestion
### Rust package maintainers

Explain how to get Rust changelogs here.

### Nuget package maintainers

Explain how to get Nuget changelogs here.

### Go package maintainers

Explain how to get Go changelogs here.

### Ruby package maintainers

Explain how to get Ruby (gem) changelogs here.
```

File: docs/usage/key-concepts/changelogs.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+Explain how to get Go changelogs here.
+
+### Ruby package maintainers
+
+Explain how to get Ruby (gem) changelogs here.
+
+## What about that `changelog.json` file?
+
+Renovate can apparently read `changelog.json` files, I don't know how that works.
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/changelogs.md:87]: Context:

- https://github.com/renovatebot/renovate/discussions/25828

**Review by rarkins — COMMENTED** [review]
@HonkingGoose I have made my edits directly to the PR. Please review and correct anything you see, and let's get it published!

File: docs/usage/key-concepts/changelogs.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+Read [Running Renovate, GitHub.com token for changelogs](../getting-started/running.md#githubcom-token-for-changelogs) to learn more.
+
+## Troubleshooting missing changelogs
+
+Follow these steps to find out why Renovate does not find a changelog:
+
+1. The datasource for this package does not support sourceUrls.
+   - If the registry fundamentally does not provide this data, then the only possibility is for it to be manually populated through PRs to Renovate's source code
+   - If the registry provides source URLs but Renovate does not support it yet, then raise a feature request with examples, or better yet a Pull Request to implement it yourself
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/changelogs.md:60]: What does `Renovate does not support it yet,` mean here? Same question with `a Pull Request to implement it yourself`.

File: docs/usage/key-concepts/changelogs.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+   - Sometimes self-hosted versions of registries don't include the full metadata compared to what the public registries do
+1. The package does not provide this information to the registry.
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/changelogs.md:62]: What does `this information` mean here?

File: docs/usage/key-concepts/changelogs.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+   - For example, occasionally `npm` packages don't have `repository` fields included
+   - For example, Docker images regularly do not have the required `LABEL` entry
+1. Renovate cannot access the source repository
+   - This is typically a concern for private repositories only
+   - Check if the token Renovate uses has access rights to the repository you need it to access
+1. Renovate cannot detect the file names or release name convention within the repository
+   - In this case an enhancement to Renovate might be needed to better detect the releases/formats, assuming it's got a reasonable convention to follow
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/changelogs.md:69]: Same question about `assuming it's got a reasonable convention to follow`. I guess you really mean something like this?

> assuming the upstream package/ecosystem has a reasonable convention that Renovate can follow

**Review by HonkingGoose — COMMENTED** [review]
We use `it` and `this` in the lists, but it's hard for me to understand what those words refer to in some cases.

Can you suggest changes so it's clear what we really mean?

File: docs/usage/key-concepts/changelogs.md (hunk 3 above)
Comment (rarkins) [inline docs/usage/key-concepts/changelogs.md:60]:
```suggestion
   - If the registry provides source URLs in its response but Renovate does not understand the required fields, then raise a feature request with examples, or better yet a Pull Request to implement support for the source URL parsing/mapping yourself
```

File: docs/usage/key-concepts/changelogs.md (hunk 4 above)
Comment (rarkins) [inline docs/usage/key-concepts/changelogs.md:62]:
```suggestion
1. The package was published without source URL information being included.
```

File: docs/usage/key-concepts/changelogs.md (hunk 5 above)
Comment (rarkins) [inline docs/usage/key-concepts/changelogs.md:69]:
```suggestion
   - In this case an enhancement to Renovate might be needed to better detect the releases/formats, assuming the package/repository has a reasonable convention to follow
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**